### PR TITLE
JIRA: USDU-168 Writing to an invalid prim crashes Unity

### PR DESCRIPTION
**Pull request by clusty**
## Purpose of this PR

USDU-168 Writing to an invalid prim crashes Unity

Bug was related to the PrimMap being out of date. 

## Testing

Unit test added

### CODE REVIEW

File: package/com.unity.formats.usd/Tests/USD.NET/StageTests.cs [hunk 1]
```diff
@@ -310,5 +310,17 @@ public static void BadPrimTest()
...
+        [Test]
+        public void WritingInvalidPrims_ShouldNotCrash()
+        {
+            var stage = pxr.UsdStage.CreateInMemory(UsdStage.InitialLoadSet.LoadNone);
+            var scene = Scene.Open(stage);
+            scene.Write("/Foo", new CubeSample());
+            scene.Stage.RemovePrim(scene.GetSdfPath("/Foo"));
+            scene.Write("/Foo", new CubeSample());
+
+            Assert.IsTrue(scene.GetPrimAtPath("/Foo").IsValid());
```
Comment (judubu) [inline package/com.unity.formats.usd/Tests/USD.NET/StageTests.cs:323]: Don't forget the scene.Close()

**Review by judubu — APPROVED** [review]
Goooood

**Comment by jcowles** [comment]
Hey @clusty do you still plan to merge this?

**Comment by clusty** [comment]
I can merge it.

**Pull request merged**